TDL-19198: Provide description of groups in README (#88)
singer-io/tap-jira: README.md

        "password": "your-jira-password",
        "base_url": "https://your-jira-domain",
        "user_agent": "<user-agent>",
        "request_timeout": 300,

>         "groups": "jira-administrators,site-admins,jira-software-users"

Reviewer: ```suggestion
        "groups": "jira-administrators, site-admins, jira-software-users"
```

Author: As shown in the [code,](https://github.com/singer-io/tap-jira/blob/TDL-19198-Provide-description-of-groups-in-README/tap_jira/streams.py#L228) when we split the 'groups' value, we are not removing extra spaces. So if we are giving 'groups' value with space as shown in the above suggestion, we are getting a list `["jira-administrators", " site-admins", " jira-software-users"]` instead of `["jira-administrators", "site-admins", "jira-software-users"]`.

Reviewer 2: I think extra space should not fail the tap. Ideally after a split we should strip() the leading spaces.

Author: In the existing code, we are not using strip, but this suggestion is right, we should ignore extra spaces, so added strip() and also update README as per code change.

     "cloud_id": "<cloud-id>",
     "refresh_token": "<refresh-token>",
     "start_date": "<i.e. 2017-12-04T19:19:32Z>",
     "request_timeout": 300,

>      "groups": "jira-administrators,site-admins,jira-software-users"

Reviewer: ```suggestion
     "groups": "jira-administrators, site-admins, jira-software-users"
```

Author: As discussed [above ](https://github.com/singer-io/tap-jira/pull/88#discussion_r881247537) we can not pass 'groups' value with extra spaces.

   can be found. For example, it might look like:
   `https://mycompany.atlassian.net`.

>    The `groups` specifies groups for users stream. `groups` is optional parameter. Default value of groups is `["jira-administrators", "jira-software-users", "jira-core-users", "jira-users", "users"]`.

Reviewer: ```suggestion
   The `groups` specifies groups for users stream. It is an optional parameter. Default value is `["jira-administrators", "jira-software-users", "jira-core-users", "jira-users", "users"]`.
```

Author: Yeah, it looks good, I have updated the comment as per suggestion.

> 

Reviewer 3: I think the default value is as per below. https://github.com/singer-io/tap-jira/blob/41be5973fa4e6d33dc7e12bbff739c53c66a8581/tap_jira/streams.py#L230-L234

Author: Yea, I have updated the groups default values as shown in streams.py
